### PR TITLE
IAST scan skip, scan schedule, scan controllers & iast_test_identifier config

### PR DESCRIPTION
**Pull request by prateeksen**
Updates in Ruby Security Configs:
- Added new security configs
```
security:
  exclude_from_iast_scan:
    api:
    http_request_parameters:
      header:
      query:
      body:
    iast_detection_category:
      insecure_settings: false
      invalid_file_access: false
      sql_injection: false
      nosql_injection: false
      ldap_injection: false
      javascript_injection: false
      command_injection: false
      xpath_injection: false
      ssrf: false
      rxss: false
  scan_schedule:
    delay: 0
    duration: 0
    schedule: ""
    always_sample_traces: false
  scan_controllers:
    iast_scan_request_rate_limit: 3600
    scan_instance_count: 0
    report_http_response_body: true
  iast_test_identifier: nil
```
- Removed security config
```
security.request.body_limit
security.detection.deserialization.enabled
security.detection.rci.enabled
security.detection.rxss.enabled
```

### CODE REVIEW

File: lib/new_relic/agent/configuration/default_source.rb [hunk 1]
```diff
@@ -2727,14 +2727,175 @@ def self.notify
...
           :description => 'The port the application is listening on. This setting is mandatory for Passenger servers. Other servers should be detected by default.'
         },
-        :'security.request.body_limit' => {
-          :default => 300,
-          :allow_nil => true,
+        :'security.exclude_from_iast_scan.api' => {
+          :default => [],
+          :public => true,
+          :type => Array,
+          :allowed_from_server => true,
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2734]:
```suggestion
          :type => Array,
          :external => true,
          :allowed_from_server => true,
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 2]
```diff
@@ -2727,14 +2727,175 @@ def self.notify
...
+          :transform => DefaultSource.method(:convert_to_list),
+          :description => 'Defines api paths you want the security agent to ignore in IAST scan, by specifying a list of patterns matching the URI you want to ignore.'
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2736]: Are wildcards `*` allows at the end of paths

File: lib/new_relic/agent/configuration/default_source.rb [hunk 3]
```diff
@@ -2727,14 +2727,175 @@ def self.notify
...
+        },
+        :'security.exclude_from_iast_scan.http_request_parameters.header' => {
+          :default => [],
+          :public => true,
+          :type => Array,
+          :allowed_from_server => true,
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2742]:
```suggestion
          :type => Array,
          :external => true,
          :allowed_from_server => true,
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 4]
```diff
@@ -2727,14 +2727,175 @@ def self.notify
...
+          :transform => DefaultSource.method(:convert_to_list),
+          :description => 'Define http request headers you want the security agent to ignore in IAST scan, by specifying a list of patterns matching the header you want to ignore.'
+        },
+        :'security.exclude_from_iast_scan.http_request_parameters.query' => {
+          :default => [],
+          :public => true,
+          :type => Array,
+          :allowed_from_server => true,
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2750]:
```suggestion
          :type => Array,
          :external => true,
          :allowed_from_server => true,
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 5]
```diff
@@ -2727,14 +2727,175 @@ def self.notify
...
+          :transform => DefaultSource.method(:convert_to_list),
+          :description => 'Defines http request query parameters you want the security agent to ignore in IAST scan, by specifying a list of patterns matching the http request query parameters you want to ignore.'
+        },
+        :'security.exclude_from_iast_scan.http_request_parameters.body' => {
+          :default => [],
+          :public => true,
+          :type => Array,
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2757]:
```suggestion
          :type => Array,
          :external => true,
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 6]
```diff
@@ -2727,14 +2727,175 @@ def self.notify
...
+          :allowed_from_server => true,
+          :transform => DefaultSource.method(:convert_to_list),
+          :description => 'Defines key values in http request body you want the security agent to ignore in IAST scan, by specifying a list of keys in the http request body you want to ignore.'
+        },
+        :'security.exclude_from_iast_scan.iast_detection_category.insecure_settings' => {
+          :default => false,
+          :external => true,
+          :public => true,
+          :type => Boolean,
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2766]:
```suggestion
          :type => Boolean,
          :external => true,
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 7]
```diff
@@ -2727,14 +2727,175 @@ def self.notify
...
+          :allowed_from_server => false,
+          :description => 'If `true`, disables RXSS (reflected cross-site scripting) detection'
+        },
+        :'security.scan_schedule.delay' => {
+          :default => 0,
+          :public => true,
+          :type => Integer,
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2845]:
```suggestion
          :type => Integer,
          :external => true,
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 8]
```diff
@@ -2727,14 +2727,175 @@ def self.notify
...
+          :default => 0,
           :public => true,
           :type => Integer,
+          :allowed_from_server => true,
+          :description => 'The duration field specifies the duration of the IAST scan in minutes. This determines how long the scan will run'
+        },
+        :'security.scan_schedule.schedule' => {
+          :default => '',
+          :public => true,
+          :type => String,
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2859]:
```suggestion
          :type => String,
          :external => true,
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 9]
```diff
@@ -2727,14 +2727,175 @@ def self.notify
...
+          :public => true,
+          :type => Boolean,
           :allowed_from_server => false,
-          :description => 'Defines the request body limit to process in security events (in KB). The default value is 300, for 300KB.'
+          :description => 'always_sample_traces permits IAST to actively gather trace data in the background, and the collected data will be used by Security Agent to perform an IAST Scan at the scheduled time'
+        },
+        :'security.scan_controllers.iast_scan_request_rate_limit' => {
+          :default => 3600,
+          :public => true,
+          :type => Integer,
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2875]:
```suggestion
          :type => Integer,
          :external => true,
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 10]
```diff
@@ -2727,14 +2727,175 @@ def self.notify
...
+          :allowed_from_server => true,
+          :description => 'Number of http request limit for ISAT scan per minute. Default is 3600. Minimum is 12 and maximum is 3600'
+        },
+        :'security.scan_controllers.scan_instance_count' => {
+          :default => 0,
+          :public => true,
+          :type => Integer,
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2882]:
```suggestion
          :type => Integer,
          :external => true,
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 11]
```diff
@@ -2727,14 +2727,175 @@ def self.notify
...
+          :allowed_from_server => true,
+          :description => 'Scan instance count controls the number of application instances for a specific entity where IAST analysis is performed'
+        },
+        :'security.scan_controllers.report_http_response_body' => {
+          :default => true,
+          :public => true,
+          :type => Boolean,
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2889]:
```suggestion
          :type => Boolean,
          :external => true,
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 12]
```diff
@@ -2727,14 +2727,175 @@ def self.notify
...
+          :transform => DefaultSource.method(:convert_to_list),
+          :description => 'Defines key values in http request body you want the security agent to ignore in IAST scan, by specifying a list of keys in the http request body you want to ignore.'
+        },
+        :'security.exclude_from_iast_scan.iast_detection_category.insecure_settings' => {
+          :default => false,
+          :external => true,
+          :public => true,
+          :type => Boolean,
+          :allowed_from_server => false,
+          :description => 'If `true`, disables Low severity insecure settings(Hash, crypto, cookie, Ranndom) detection'
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2768]: Does this still make sense? Is the list of examples just a few examples or everything? I'm not familiar with those settings

```suggestion
          :description => 'If `true`, disables detection of low-severity insecure settings (e.g., hash, crypto, cookie, random generators)'
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 13]
```diff
@@ -2727,14 +2727,175 @@ def self.notify
...
+        },
+        :'security.exclude_from_iast_scan.iast_detection_category.invalid_file_access' => {
+          :default => false,
+          :external => true,
+          :public => true,
+          :type => Boolean,
+          :allowed_from_server => false,
+          :description => 'If `true`, disables file access related IAST detections'
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2776]: Can we list the file access-related ISAT detections?

File: lib/new_relic/agent/configuration/default_source.rb [hunk 14]
```diff
@@ -2727,14 +2727,175 @@ def self.notify
...
+        },
+        :'security.exclude_from_iast_scan.iast_detection_category.sql_injection' => {
+          :default => false,
+          :external => true,
+          :public => true,
+          :type => Boolean,
+          :allowed_from_server => false,
+          :description => 'If `true`, disables SQL injection detection'
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2784]:
```suggestion
          :description => 'If `true`, disables SQL injection detection in IAST scans.'
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 15]
```diff
@@ -2727,14 +2727,175 @@ def self.notify
...
+          :allowed_from_server => false,
+          :description => 'If `true`, disables NOSQL injection detection'
+        },
+        :'security.exclude_from_iast_scan.iast_detection_category.ldap_injection' => {
+          :default => false,
+          :external => true,
+          :public => true,
+          :type => Boolean,
+          :allowed_from_server => false,
+          :description => 'If `true`, disables LDAP injection detection'
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2800]:
```suggestion
          :description => 'If `true`, disables LDAP injection detection in IAST scans.'
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 16]
```diff
@@ -2727,14 +2727,175 @@ def self.notify
...
+        },
+        :'security.exclude_from_iast_scan.iast_detection_category.javascript_injection' => {
+          :default => false,
+          :external => true,
+          :public => true,
+          :type => Boolean,
+          :allowed_from_server => false,
+          :description => 'If `true`, disables javascript injection detection'
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2808]:
```suggestion
          :description => 'If `true`, disables javascript injection detection in IAST scans.'
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 17]
```diff
@@ -2727,14 +2727,175 @@ def self.notify
...
+        },
+        :'security.exclude_from_iast_scan.iast_detection_category.command_injection' => {
+          :default => false,
+          :external => true,
+          :public => true,
+          :type => Boolean,
+          :allowed_from_server => false,
+          :description => 'If `true`, disables system command injection detection'
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2816]:
```suggestion
          :description => 'If `true`, disables system command injection detection in IAST scans.'
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 18]
```diff
@@ -2727,14 +2727,175 @@ def self.notify
...
+        },
+        :'security.exclude_from_iast_scan.iast_detection_category.xpath_injection' => {
+          :default => false,
+          :external => true,
+          :public => true,
+          :type => Boolean,
+          :allowed_from_server => false,
+          :description => 'If `true`, disables XPATH injection detection'
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2824]:
```suggestion
          :description => 'If `true`, disables XPATH injection detection in IAST scans.'
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 19]
```diff
@@ -2727,14 +2727,175 @@ def self.notify
...
+        },
+        :'security.exclude_from_iast_scan.iast_detection_category.ssrf' => {
+          :default => false,
+          :external => true,
+          :public => true,
+          :type => Boolean,
+          :allowed_from_server => false,
+          :description => 'If `true`, disables SSRF(Server side reuqest forgery) detection'
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2832]:
```suggestion
          :description => 'If `true`, disables Sever-Side Request Forgery (SSRF) detection in IAST scans.'
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 20]
```diff
@@ -2727,14 +2727,175 @@ def self.notify
...
+        },
+        :'security.exclude_from_iast_scan.iast_detection_category.rxss' => {
+          :default => false,
+          :external => true,
+          :public => true,
+          :type => Boolean,
+          :allowed_from_server => false,
+          :description => 'If `true`, disables RXSS (reflected cross-site scripting) detection'
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2840]:
```suggestion
          :description => 'If `true`, disables Reflected Cross-Site Scripting (RXSS) detection in IAST scans.'
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 21]
```diff
@@ -2727,14 +2727,175 @@ def self.notify
...
+        },
+        :'security.scan_schedule.delay' => {
+          :default => 0,
+          :public => true,
+          :type => Integer,
+          :allowed_from_server => true,
+          :description => 'The delay field indicated time in minutes before the IAST scan starts after the application starts'
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2847]:
```suggestion
          :description => 'Specifies the delay time (in minutes) before the IAST scan begins after the application starts.'
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 22]
```diff
@@ -2727,14 +2727,175 @@ def self.notify
...
+        },
+        :'security.scan_schedule.duration' => {
+          :default => 0,
           :public => true,
           :type => Integer,
+          :allowed_from_server => true,
+          :description => 'The duration field specifies the duration of the IAST scan in minutes. This determines how long the scan will run'
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2854]:
```suggestion
          :description => 'Specifies the length of time (in minutes) that the IAST scan will run.'
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 23]
```diff
@@ -2727,14 +2727,175 @@ def self.notify
...
+        },
+        :'security.scan_schedule.schedule' => {
+          :default => '',
+          :public => true,
+          :type => String,
+          :allowed_from_server => true,
+          :description => 'The schedule field specifies a cron expression that defines when the IAST scan should run',
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2861]:
```suggestion
          :description => 'Specifies a cron expression that sets when the IAST scan should run.',
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 24]
```diff
@@ -2727,14 +2727,175 @@ def self.notify
...
+          :dynamic_name => true
+        },
+        :'security.scan_schedule.always_sample_traces' => {
+          :default => false,
           :external => true,
+          :public => true,
+          :type => Boolean,
           :allowed_from_server => false,
-          :description => 'Defines the request body limit to process in security events (in KB). The default value is 300, for 300KB.'
+          :description => 'always_sample_traces permits IAST to actively gather trace data in the background, and the collected data will be used by Security Agent to perform an IAST Scan at the scheduled time'
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2870]:
```suggestion
          :description => 'If `true`, allows IAST to continuously gather trace data in the background. Collected data will be used by Security Agent to perform an IAST Scan at the scheduled time.'
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 25]
```diff
@@ -2727,14 +2727,175 @@ def self.notify
...
+        },
+        :'security.scan_controllers.iast_scan_request_rate_limit' => {
+          :default => 3600,
+          :public => true,
+          :type => Integer,
+          :allowed_from_server => true,
+          :description => 'Number of http request limit for ISAT scan per minute. Default is 3600. Minimum is 12 and maximum is 3600'
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2877]:
```suggestion
          :description => 'Sets the maximum number of HTTP requests allowed for the ISAT scan per minute. Any Integer between 12 and 3600 is valid. The default value is 3600.'
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 26]
```diff
@@ -2727,14 +2727,175 @@ def self.notify
...
+        },
+        :'security.scan_controllers.scan_instance_count' => {
+          :default => 0,
+          :public => true,
+          :type => Integer,
+          :allowed_from_server => true,
+          :description => 'Scan instance count controls the number of application instances for a specific entity where IAST analysis is performed'
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2884]:
```suggestion
          :description => The number of application instances for a specific entity on which IAST analysis is performed.'
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 27]
```diff
@@ -2727,14 +2727,175 @@ def self.notify
...
+        },
+        :'security.scan_controllers.report_http_response_body' => {
+          :default => true,
+          :public => true,
+          :type => Boolean,
+          :allowed_from_server => true,
+          :description => 'report_http_response_body allows user to enable/disable sending of HTTP responses body, Disabling this would also disable RXSS vulnerability detection'
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2891]:
```suggestion
          :description => 'If `true`, enables the sending of HTTP responses bodies. Disabling this also disables Reflected Cross-Site Scripting (RXSS) vulnerability detection.'
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 28]
```diff
@@ -2727,14 +2727,175 @@ def self.notify
...
+        },
+        :'security.iast_test_identifier' => {
+          :default => nil,
+          :public => true,
+          :type => String,
+          :allowed_from_server => true,
+          :description => 'This will allow users to run IAST for CI/CD'
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2898]: I'm not sure what an example input would be here- is there any way to expand on the description?

**Review by hannahramadan — CHANGES_REQUESTED** [review]
Thanks for accepting the suggested changes!

Can we check on the failing unit tests? I also have a few open comments on some of the configs that I'm still wondering about.

**Comment by prateeksen** [comment]
@hannahramadan Rest of the open comments and failing UTs, I will look at those today.

File: lib/new_relic/agent/configuration/default_source.rb (hunk 13 above)
Comment (prateeksen) [reply lib/new_relic/agent/configuration/default_source.rb]: @hannahramadan Description updated in this [commit](https://github.com/newrelic/newrelic-ruby-agent/pull/2934/commits/0eb564c474637ce6994716f70a45a29a7d331501)

File: lib/new_relic/agent/configuration/default_source.rb (hunk 28 above)
Comment (prateeksen) [reply lib/new_relic/agent/configuration/default_source.rb]: @hannahramadan Updated description in [this](https://github.com/newrelic/newrelic-ruby-agent/pull/2934/commits/4204fa5f3f692cd8bb28c69f082a3f9d345ba30b) commit

File: lib/new_relic/agent/configuration/default_source.rb (hunk 2 above)
Comment (prateeksen) [reply lib/new_relic/agent/configuration/default_source.rb]: @hannahramadan No, Wild cards are not allowed, The regex pattern should provide a complete match. I have updated the description in with an example in [this](https://github.com/newrelic/newrelic-ruby-agent/pull/2934/commits/df5c3affed130ffeda11308a1b917916f9a1666c) commit.

**Comment by prateeksen** [comment]
> Thanks for accepting the suggested changes!
> 
> Can we check on the failing unit tests? I also have a few open comments on some of the configs that I'm still wondering about.

@hannahramadan I have resolved all the comments, Could you please re-review.

File: lib/new_relic/agent/configuration/default_source.rb [hunk 29]
```diff
@@ -2703,47 +2703,196 @@ def self.notify
...
+          :description => 'Define http request headers you want the security agent to ignore in IAST scan, by specifying a list of patterns matching the header you want to ignore.'
+        },
+        :'security.exclude_from_iast_scan.http_request_parameters.query' => {
+          :default => [],
+          :public => true,
+          :type => Array,
+          :external => true,
+          :allowed_from_server => true,
+          :transform => DefaultSource.method(:convert_to_list),
+          :description => 'Defines http request query parameters you want the security agent to ignore in IAST scan, by specifying a list of patterns matching the http request query parameters you want to ignore.'
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2740]:
```suggestion
          :description => 'An array of HTTP request query parameters the security agent should ignore in IAST scans. The array should specify a list of patterns matching the HTTP request query parameters to ignore.'
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 30]
```diff
@@ -2703,47 +2703,196 @@ def self.notify
...
+        },
+        :'security.exclude_from_iast_scan.http_request_parameters.body' => {
+          :default => [],
+          :public => true,
+          :type => Array,
+          :external => true,
+          :allowed_from_server => true,
+          :transform => DefaultSource.method(:convert_to_list),
+          :description => 'Defines key values in http request body you want the security agent to ignore in IAST scan, by specifying a list of keys in the http request body you want to ignore.'
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2749]:
```suggestion
          :description => 'An array of HTTP request body keys the security agent should ignore in IAST scans.'
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 31]
```diff
@@ -2703,47 +2703,196 @@ def self.notify
...
+          :dynamic_name => true
+        },
+        :'security.scan_schedule.always_sample_traces' => {
+          :default => false,
+          :external => true,
+          :public => true,
+          :type => Boolean,
           :allowed_from_server => false,
-          :description => 'Defines the request body limit to process in security events (in KB). The default value is 300, for 300KB.'
+          :description => 'If `true`, allows IAST to continuously gather trace data in the background. Collected data will be used by Security Agent to perform an IAST Scan at the scheduled time.'
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2862]:
```suggestion
          :description => 'If `true`, allows IAST to continuously gather trace data in the background. Collected data will be used by the security agent to perform an IAST scan at the scheduled time.'
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 32]
```diff
@@ -2703,47 +2703,196 @@ def self.notify
...
-        :'security.detection.rxss.enabled' => {
-          :default => true,
+        :'security.exclude_from_iast_scan.iast_detection_category.invalid_file_access' => {
+          :default => false,
           :external => true,
           :public => true,
           :type => Boolean,
           :allowed_from_server => false,
-          :description => 'If `true`, enables RXSS (reflected cross-site scripting) detection'
+          :description => 'If `true`, disables file operation related IAST detections(File Access & Application integrity violation)'
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2765]:
```suggestion
          :description => 'If `true`, disables file operation-related IAST detections (File Access & Application integrity violation)'
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 33]
```diff
@@ -2703,47 +2703,196 @@ def self.notify
...
+          :description => 'Defines key values in http request body you want the security agent to ignore in IAST scan, by specifying a list of keys in the http request body you want to ignore.'
+        },
+        :'security.exclude_from_iast_scan.iast_detection_category.insecure_settings' => {
+          :default => false,
           :external => true,
           :public => true,
           :type => Boolean,
           :allowed_from_server => false,
-          :description => 'If `true`, enables RCI (remote code injection) detection'
+          :description => 'If `true`, disables detection of low-severity insecure settings (e.g., hash, crypto, cookie, random generators, trust boundary).'
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2757]:
```suggestion
          :description => 'If `true`, disables the detection of low-severity insecure settings (e.g., hash, crypto, cookie, random generators, trust boundary).'
```

File: lib/new_relic/agent/configuration/default_source.rb [hunk 34]
```diff
@@ -2703,47 +2703,196 @@ def self.notify
...
-        :'security.request.body_limit' => {
-          :default => 300,
-          :allow_nil => true,
+        :'security.exclude_from_iast_scan.iast_detection_category.javascript_injection' => {
+          :default => false,
+          :external => true,
+          :public => true,
+          :type => Boolean,
+          :allowed_from_server => false,
+          :description => 'If `true`, disables javascript injection detection in IAST scans.'
```
Comment (hannahramadan) [inline lib/new_relic/agent/configuration/default_source.rb:2797]:
```suggestion
          :description => 'If `true`, disables Javascript injection detection in IAST scans.'
```

**Review by hannahramadan — DISMISSED** [review]

**Review by hannahramadan — APPROVED** [review]

**Pull request merged**